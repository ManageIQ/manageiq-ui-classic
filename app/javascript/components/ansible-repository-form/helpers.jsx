const GitUrlParse = require('git-url-parse');

export const urlValidation = (values) => {
  const errors = {};
  var error = false;
  try {
    const parsed = values.scm_url ? GitUrlParse(values.scm_url) : {};
    if ((parsed.protocol !== 'http'
          && parsed.protocol !== 'https'
          && parsed.protocol !== 'file'
          && parsed.protocol !== 'ssh')
          || parsed.pathname === '') {
      error = true;
    }
    if (parsed.protocols.length > 2 || (parsed.protocols.length === 1 &&  parsed.protocols[0] !== parsed.protocol)) {
      error = true;
    }
  } catch (err) {
    error = true;
  }
  if (error) {
    errors.scm_url = __('URL must include a protocol (http://, https:// or file://) with path or be a valid SSH path (user@server:path or ssh://user@address:port/path)');
  }
  return errors;
};
