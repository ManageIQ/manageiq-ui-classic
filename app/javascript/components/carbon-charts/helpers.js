export const sampleData = [{
  group: 'Dataset 1',
  key: '1',
  value: 100,
},
{
  group: 'Dataset 1',
  key: '2',
  value: 400,
},
{
  group: 'Dataset 1',
  key: '4',
  value: 500,
},
{
  group: 'Dataset 1',
  key: '3',
  value: 200,
},
{
  group: 'Dataset 1',
  key: '5',
  value: 150,
},
{
  group: 'Dataset 2',
  key: '1',
  value: 280,
},
{
  group: 'Dataset 2',
  key: '2',
  value: 350,
},
{
  group: 'Dataset 2',
  key: '3',
  value: 700,
},
{
  group: 'Dataset 2',
  key: '5',
  value: 600,
},
{
  group: 'Dataset 2',
  key: '5',
  value: 200,
},
{
  group: 'Dataset 3',
  key: '4',
  value: 100,
},
{
  group: 'Dataset 3',
  key: '2',
  value: 500,
},
{
  group: 'Dataset 3',
  key: '1',
  value: 200,
}];

export const pieData = [{
  group: 'Dataset 1',
  key: '1',
  value: 100,
},
{
  group: 'Dataset 2',
  key: '1',
  value: 280,
},
{
  group: 'Dataset 3',
  key: '4',
  value: 100,
},
{
  group: 'Dataset 4',
  key: '2',
  value: 500,
}];

// convert report data to carbon chart data format.
export const getConvertedData = (data) => {
  if (data && data.data.columns && data.data && data.miq && data.miq.category_table && data.miq.name_table) {
    const columnsData = data.data.columns;
    const dataGroups = data.miq.category_table;
    const rowsData = data.miq.name_table;
    const arr = [];
    if (columnsData.length > 0) {
      columnsData.forEach((items) => {
        items.forEach((item, i) => {
          const obj = {};
          if (i !== 0 && rowsData[items[0]]) {
            obj.group = rowsData[items[0]];
            obj.key = dataGroups[i - 1];
            obj.value = item;
            arr.push(obj);
          }
        });
      });
    }
    return arr;
  }
  return [];
};

export const getLineConvertedData = (data) => {
  const columnsData = data.data.columns;
  const dataGroups = data.miq.category_table;
  const rowsData = data.miq.name_table;
  const arr = [];
  columnsData.forEach((items) => {
    items.forEach((item, i) => {
      const obj = {};
      if (i !== 0 && rowsData[items[0]]) {
        obj.group = rowsData[items[0]];
        obj.key = new Date(dataGroups[i - 1]);
        obj.value = item;
        arr.push(obj);
      }
    });
  });
  return arr;
};
