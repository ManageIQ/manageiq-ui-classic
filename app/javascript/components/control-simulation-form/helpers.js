// Helper Functions

var showPassedPolicies = true;
var showFailedPolicies = true;
var showOutOfScopeItems = true;

const convertApi = (apiData, showPassedPolicies, showFailedPolicies, showOutOfScopeSelected) => {
  showPassedPolicies = showPassedPolicies;
  showFailedPolicies = showFailedPolicies;
  showOutOfScopeItems = showOutOfScopeSelected;

  if (apiData === undefined) {
    return '';
  }
  console.log('api Data');
  console.log(apiData);

  const convToString = [{
    key: 'root',
    text: apiData.name,
    icon: 'fa fa-star',
    class: 'no-cursor',
    selectable: false,
    state: {
      expanded: true,
    },
    nodes: findAllVms(apiData.task_results),
  }];
  return convToString;
};

const findAllVms = (allVms) => {
  // TODO: test this, there are cases with more than one vm (By Clusters -> Cluster)
  const holdAllVms = [];

  allVms.forEach((vmData) => {
    const vmObj = {
      key: `xx-${vmData.id}`,
      text: `\u003cstrong\u003eVM:\u003c/strong\u003e ${vmData.name}`,
      icon: 'pficon pficon-virtual-machine',
      class: 'no-cursor',
      selectable: false,
      nodes: findAllProfiles(vmData.profiles, `xx-${vmData.id}`),
      state: {
        expanded: false,
      },
    };
    holdAllVms.push(vmObj);
  });
  return holdAllVms;
};

const findAllProfiles = (allProfiles, upperKey) => {
  const holdAllPolicies = [];

  allProfiles.forEach((profileData) => {
    const profileObj = {
      key: `${upperKey}_xx-${profileData.id}`,
      text: `\u003cstrong\u003eProfile:\u003c/strong\u003e ${profileData.name}`,
      icon: determineIcon(profileData.result),
      class: 'no-cursor',
      selectable: false,
      nodes: findAllPolicies(profileData.policies, `${upperKey}_xx-${profileData.id}`),
      state: {
        expanded: false,
      },
    };
    holdAllPolicies.push(profileObj);
  });
  return holdAllPolicies;
};

const findAllPolicies = (allPolicies, upperKey) => {
  const holdAllPolicies = [];
  allPolicies.forEach((policyData) => {
    // NOTE: you cannot give the tree a {} option it will break
    const nodes = [];
    if (policyData.scope) {
      nodes.push(findAllScope(policyData.scope, `${upperKey}_xx-${policyData.id}`));
    }
    if (policyData.conditions.length !== 0) {
      nodes.push(findAllConditions(policyData.conditions, `${upperKey}_xx-${policyData.id}`));
    }
    if (policyData.actions.length !== 0) {
      nodes.push(findAllActions(policyData.actions, `${upperKey}_xx-${policyData.id}`));
    }
    const policyObj = {
      key: `${upperKey}_xx-${policyData.id}`,
      text: `\u003cstrong\u003ePolicy:\u003c/strong\u003e ${policyData.description}`,
      icon: determineIcon(policyData.result),
      class: 'no-cursor',
      selectable: false,
      nodes,
      state: {
        expanded: false,
      },
    };

    // NOTE: policyData.result can be: deny || allow || N/A
    if (policyData.result === 'deny' && showFailedPolicies === true) {
      holdAllPolicies.push(policyObj);
    } else if (policyData.result === 'allow' && showPassedPolicies === true) {
      holdAllPolicies.push(policyObj);
    } else if (policyData.result === 'N/A' && showOutOfScopeItems === true) {
      holdAllPolicies.push(policyObj);
    }
  });
  return holdAllPolicies;
};

const findAllConditions = (allConditions, upperKey) => {
  let holdAllConditions = {};

  allConditions.forEach((conditionData) => {
    const conditionObj = {
      key: `${upperKey}_xx-${conditionData.id}`,
      text: `\u003cstrong\u003eCondition:\u003c/strong\u003e ${conditionData.description}`,
      icon: determineIcon(conditionData.result),
      class: 'no-cursor',
      selectable: false,
      nodes: [findAllExpressions(conditionData.expression, `${upperKey}_xx-${conditionData.id}`)],
      state: {
        expanded: false,
      },
    };
    holdAllConditions = conditionObj;
  });
  return holdAllConditions;
};

const findAllActions = (allActions, upperKey) => {
  let holdAllActions = {};

  allActions.forEach((actionData) => {
    const actionObj = {
      key: `${upperKey}_xx-${actionData.id}`,
      text: `\u003cstrong\u003eAction:\u003c/strong\u003e ${actionData.description}`,
      icon: determineIcon(actionData.result),
      class: 'no-cursor',
      selectable: false,
      state: {
        expanded: false,
      },
    };
    holdAllActions = actionObj;
  });
  return holdAllActions;
};

const findAllExpressions = (allExpressions, upperKey) => {
  const dynamicallyBuildText = createSentance(allExpressions, '');
  const holdExpression = {
    key: `${upperKey}_xx-`, // NOTE: this doesnt have an id
    text: `\u003cstrong\u003eExpression:\u003c/strong\u003e ${determineTextColor(allExpressions.result, dynamicallyBuildText)}`,
    tooltip: dynamicallyBuildText,
    icon: determineIcon(allExpressions.result),
    class: 'no-cursor',
    selectable: false,
    state: {
      expanded: false,
    },
  };
  return holdExpression;
};

const findAllScope = (allScope, upperKey) => {
  const dynamicallyBuildText = createSentance(allScope, '');
  const holdScope = {
    key: `${upperKey}_xx-`, // NOTE: this doesnt have an id
    text: `\u003cstrong\u003eScope:\u003c/strong\u003e ${determineTextColor(allScope.result, dynamicallyBuildText)}`,
    tooltip: dynamicallyBuildText,
    icon: determineIcon(allScope.result), // TODO: Is scope always red x ?
    class: 'no-cursor',
    selectable: false,
    state: {
      expanded: false,
    },
  };
  return holdScope;
};

export { convertApi };

// ==================================================================== //

const createSentance = (object, runOnSentence) => {
  // TODO: Test more
  let result = runOnSentence;
  // var useAnd = result == "" ? "" : " && ";
  for (const key in object) {
    if (key == 'result') {
      return result;
    } if (object[key].field != undefined) {
      result = `${result + (result === '' ? '' : ' and ') + object[key].field} ${key} ${object[key].value}`;
    } else {
      result = result + (result === '' ? '' : ' and ') + createSentance(object[key], runOnSentence);
    }
  }
  return result;
};

const determineIcon = (resultValue) => {
  // "icon" :
  // "fa fa-ban"                    <- grey crossed circle
  // "pficon pficon-ok"             <- green check circle
  // "pficon pficon-error-circle-o" <- red x circle
  switch (resultValue) {
    case 'deny':
      return 'pficon pficon-error-circle-o';
    case 'N/A':
      return 'fa fa-ban';
    case 'allow':
      return 'pficon pficon-ok';
    case true: // this happens in the scope / expression level
      return 'pficon pficon-ok'; // "fa fa-bell";
    case false:
      return 'pficon pficon-error-circle-o'; // "fa fa-bell-o";
    default:
      return 'fa fa-ban'; // "fa fa-bell-slash";
  }
};

const determineTextColor = (resultValue, text) => {
  switch (resultValue) {
    case true: // this happens in the scope / expression level
      return `\u003cfont color="green"\u003e${text}\u003c/font\u003e`;
    case false:
      return `\u003cfont color="red"\u003e${text}\u003c/font\u003e`;
    default:
      return text;
  }
};
