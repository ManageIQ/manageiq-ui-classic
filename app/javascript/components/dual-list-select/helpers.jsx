/* *********** HELPERS FOR FORMS *********** */

/* Get keys of options array
  Options:
    [
      {key: 'key', label: 'label'},
      {key: 'key1', label: 'label1'},
    ]
  returns ['key', 'key1']
*/
export const getKeys = value => (Array.isArray(value) ? value.map(({ key }) => key) : []);

/* Returns array of options excluding value options */
export const filterOptions = (options, value) => options.filter(({ key }) => !value.includes(key));

/* Returns items from newValue which are not in value */
export const filterValues = (newValue, value) => newValue.filter(item => !value.includes(item));
