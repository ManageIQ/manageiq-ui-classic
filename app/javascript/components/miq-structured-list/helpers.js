/* eslint-disable no-eval */
import { TreeViewRedux } from '../tree-view';
import CatalogResource from '../data-tables/catalog-resource';
import MiqDataTable from '../miq-data-table';

export const InputTypes = {
  TEXTAREA: 'text_area',
  CHECKBOX: 'checkbox',
  COMPONENT: 'component',
  DROPDOWN: 'dropdown',
  CODEMIRROR: 'code_mirror',
  MARKDOWN: 'markdown',
};

export const DynamicReactComponents = {
  TREE_VIEW_REDUX: TreeViewRedux,
  CATALOG_RESOURCE: CatalogResource,
  MIQ_DATA_TABLE: MiqDataTable,
};

const dataType = (data) => (data ? data.constructor.name.toString() : undefined);

export const isObject = (item) => dataType(item) === 'Object';

export const isArray = (item) => dataType(item) === 'Array';

export const isSubItem = (item) => item.sub_items && item.sub_items.length > 0;

/** When the props named mode includes 'miq_summary', then then summary contains row/cell click events. */
export const hasClickEvents = (mode) => mode.includes('miq_summary');

export const hasInput = (item) => item && Object.keys(item).includes('input');

export const rowClickEvent = (rows, index) => {
  const item = rows[index];
  if (item && item.onclick) {
    miqSparkleOn();
    eval(item.onclick);
    miqSparkleOff();
  }
};
