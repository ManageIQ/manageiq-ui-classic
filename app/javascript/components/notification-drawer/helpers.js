export const newCountText = function(count) {
  return sprintf(n__('%d New', '%d New', count), count);
};

export const getNotficationStatusIconName = notification => (
  {
    info: 'info',
    error: 'error',
    danger: 'error',
    warning: 'warning',
    success: 'success',
    ok: 'success',
  }[notification.type] || '');

export const viewDetails = function(notification) {
  window.location.href = `/restful_redirect?${$.param(notification.data.link)}`;
};
