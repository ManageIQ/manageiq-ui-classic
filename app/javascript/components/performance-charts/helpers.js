export const getYAxisValue = (format, value) => {
  // eslint-disable-next-line no-useless-escape
  if (format) {
    const tmp = /^([0-9\,\.]+)(.*)/.exec(ManageIQ.charts.formatters[format.function].c3(format.options)(value));
    return [`${numeral(tmp[1]).value()}${tmp[2]}`];
  }
  return value;
};
