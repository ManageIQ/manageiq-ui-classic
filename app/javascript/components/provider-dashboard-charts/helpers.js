import moment from 'moment';

export const getConvertedData = (data, units) => {
  if (data && data.xData && data.yData) {
    const columnsData = data.xData;
    const rowsData = data.yData;
    const arr = [];
    if (columnsData.length > 0) {
      columnsData.forEach((item, i) => {
        const obj = {};
        obj.group = units;
        obj.date = moment(item).tz(ManageIQ.timezone || 'UTC').format('MM/DD/YYYY HH:mm:ss z')
        obj.value = rowsData[i];
        arr.push(obj);
      });
    }
    return arr;
  }
  return [];
};

export const getLatestValue = (data) => {
  if (data && data.yData) {
    return data.yData[data.yData.length - 1];
  }
  return 0;
};

export const getHeatMapData = (data) => {
  if (data) {
    const arr = [];
    data.forEach((item) => {
      const obj = {};
      obj.percent = item.percent;
      obj.total = item.total;
      obj.provider = item.provider;
      obj.node = item.node;
      obj.value = item.percent * 100;
      obj.unit = item.unit;
      arr.push(obj);
    });
    return arr;
  }
  return [];
};

export const getPodsData = (data, createdlabel, deletedLabel) => {
  if (data && data.xData && data.yCreated && data.yDeleted) {
    const columnsData = data.xData;
    const createdData = data.yCreated;
    const deletedData = data.yDeleted;
    const arr = [];
    if (createdData.length > 0) {
      createdData.forEach((item, i) => {
        const obj = {};
        obj.group = createdlabel;
        obj.date = columnsData[i];
        obj.value = item;
        arr.push(obj);
      });
    }
    if (deletedData.length > 0) {
      deletedData.forEach((item, i) => {
        const obj = {};
        obj.group = deletedLabel;
        obj.date = columnsData[i];
        obj.value = item;
        arr.push(obj);
      });
    }
    return arr;
  }
  return [];
};
