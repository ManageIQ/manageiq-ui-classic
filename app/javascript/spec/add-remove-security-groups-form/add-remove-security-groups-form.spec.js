import React from 'react';
import toJson from 'enzyme-to-json';
import fetchMock from 'fetch-mock';
import { act } from 'react-dom/test-utils';
import { mount } from '../helpers/mountForm';
import AddRemoveSecurityGroupForm from '../../components/vm-cloud-add-remove-security-group-form';

require('../helpers/miqSparkle.js');
require('../helpers/miqAjaxButton.js');

describe('Add/remove security groups form component', () => {
  let submitSpy;
  beforeEach(() => {
    submitSpy = jest.spyOn(window, 'miqAjaxButton');
  });
  afterEach(() => {
    fetchMock.reset();
    fetchMock.restore();
    submitSpy.mockRestore();
  });
  it('should render add security group form', () => {
    const wrapper = mount(<AddRemoveSecurityGroupForm recordId="1850" redirectURL="/vm_cloud/explorer" isAdd />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render remove security group form', () => {
    const wrapper = mount(<AddRemoveSecurityGroupForm recordId="1850" redirectURL="/vm_cloud/explorer" isAdd={false} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should add security group', async(done) => {
    const currentSecurityGroups = {
      resources: [
        { href: 'http://localhost:3000/api/instances/1850/security_groups/84', id: '84', name: 'default' },
      ],
    };
    const securityGroups = {
      resources: [
        { href: 'http://localhost:3000/api/cloud_tenants/2/security_groups/84', id: '84', name: 'default' },
        { href: 'http://localhost:3000/api/cloud_tenants/2/security_groups/85', id: '85', name: 'loic' },
        { href: 'http://localhost:3000/api/cloud_tenants/2/security_groups/284', id: '284', name: 'docs-test' },
        { href: 'http://localhost:3000/api/cloud_tenants/2/security_groups/88', id: '88', name: 'dave' },
        { href: 'http://localhost:3000/api/cloud_tenants/2/security_groups/268', id: '268', name: 'Test Loic' },
        { href: 'http://localhost:3000/api/cloud_tenants/2/security_groups/269', id: '269', name: 'test_loic_1' },
      ],
    };
    const saveObject = {
      name: 'loic',
      action: 'add',
    };
    fetchMock.getOnce('/api/instances/1850/security_groups?expand=resources&attributes=id,name', currentSecurityGroups);
    fetchMock.getOnce('/api/instances/1850', { cloud_tenant_id: '2' });
    fetchMock.getOnce('/api/cloud_tenants/2/security_groups?expand=resources&attributes=id,name', securityGroups);
    fetchMock.postOnce('/api/instances/1850/security_groups/', saveObject);
    const wrapper = mount(<AddRemoveSecurityGroupForm recordId="1850" redirectURL="/vm_cloud/explorer" isAdd />);
    expect(toJson(wrapper)).toMatchSnapshot();
    done();
  });
  it('should remove security group', async(done) => {
    const currentSecurityGroups = {
      resources: [
        { href: 'http://localhost:3000/api/instances/1850/security_groups/84', id: '84', name: 'default' },
      ],
    };
    const saveObject = {
      name: 'default',
      action: 'remove',
    };
    fetchMock.getOnce('/api/instances/1850/security_groups?expand=resources&attributes=id,name', currentSecurityGroups);
    fetchMock.postOnce('/api/instances/1850/security_groups/', saveObject);
    let wrapper;
    await act(async() => {
      wrapper = mount(<AddRemoveSecurityGroupForm recordId="1850" redirectURL="/vm_cloud/explorer" isAdd={false} />);
    });
    wrapper.update();
    expect(toJson(wrapper)).toMatchSnapshot();
    done();
  });
});
