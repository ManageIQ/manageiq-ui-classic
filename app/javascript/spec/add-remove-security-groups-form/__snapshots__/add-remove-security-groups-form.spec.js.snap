// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add/remove security groups form component should add security group 1`] = `
<Provider
  store={
    Object {
      "asyncReducers": Object {
        "FormButtons": [Function],
        "miqCustomTabReducer": [Function],
        "notificationReducer": [Function],
      },
      "dispatch": [Function],
      "getState": [Function],
      "injectReducers": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <AddRemoveSecurityGroupForm
    isAdd={true}
    recordId="1850"
    redirectURL="/vm_cloud/explorer"
  >
    <Loading
      active={true}
      className="export-spinner"
      description="Active loading indicator"
      small={true}
      withOverlay={false}
    >
      <div
        aria-atomic="true"
        aria-live="assertive"
        className="export-spinner bx--loading bx--loading--small"
      >
        <label
          className="bx--visually-hidden"
          id="loading-id-3"
        >
          Active loading indicator
        </label>
        <svg
          className="bx--loading__svg"
          viewBox="0 0 100 100"
        >
          <title>
            Active loading indicator
          </title>
          <circle
            className="bx--loading__background"
            cx="50%"
            cy="50%"
            r="44"
          />
          <circle
            className="bx--loading__stroke"
            cx="50%"
            cy="50%"
            r="44"
          />
        </svg>
      </div>
    </Loading>
  </AddRemoveSecurityGroupForm>
</Provider>
`;

exports[`Add/remove security groups form component should remove security group 1`] = `
<Provider
  store={
    Object {
      "asyncReducers": Object {
        "FormButtons": [Function],
        "miqCustomTabReducer": [Function],
        "notificationReducer": [Function],
      },
      "dispatch": [Function],
      "getState": [Function],
      "injectReducers": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <AddRemoveSecurityGroupForm
    isAdd={false}
    recordId="1850"
    redirectURL="/vm_cloud/explorer"
  >
    <Loading
      active={true}
      className="export-spinner"
      description="Active loading indicator"
      small={true}
      withOverlay={false}
    >
      <div
        aria-atomic="true"
        aria-live="assertive"
        className="export-spinner bx--loading bx--loading--small"
      >
        <label
          className="bx--visually-hidden"
          id="loading-id-4"
        >
          Active loading indicator
        </label>
        <svg
          className="bx--loading__svg"
          viewBox="0 0 100 100"
        >
          <title>
            Active loading indicator
          </title>
          <circle
            className="bx--loading__background"
            cx="50%"
            cy="50%"
            r="44"
          />
          <circle
            className="bx--loading__stroke"
            cx="50%"
            cy="50%"
            r="44"
          />
        </svg>
      </div>
    </Loading>
  </AddRemoveSecurityGroupForm>
</Provider>
`;

exports[`Add/remove security groups form component should render add security group form 1`] = `
<Provider
  store={
    Object {
      "asyncReducers": Object {
        "FormButtons": [Function],
        "miqCustomTabReducer": [Function],
        "notificationReducer": [Function],
      },
      "dispatch": [Function],
      "getState": [Function],
      "injectReducers": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <AddRemoveSecurityGroupForm
    isAdd={true}
    recordId="1850"
    redirectURL="/vm_cloud/explorer"
  >
    <Loading
      active={true}
      className="export-spinner"
      description="Active loading indicator"
      small={true}
      withOverlay={false}
    >
      <div
        aria-atomic="true"
        aria-live="assertive"
        className="export-spinner bx--loading bx--loading--small"
      >
        <label
          className="bx--visually-hidden"
          id="loading-id-1"
        >
          Active loading indicator
        </label>
        <svg
          className="bx--loading__svg"
          viewBox="0 0 100 100"
        >
          <title>
            Active loading indicator
          </title>
          <circle
            className="bx--loading__background"
            cx="50%"
            cy="50%"
            r="44"
          />
          <circle
            className="bx--loading__stroke"
            cx="50%"
            cy="50%"
            r="44"
          />
        </svg>
      </div>
    </Loading>
  </AddRemoveSecurityGroupForm>
</Provider>
`;

exports[`Add/remove security groups form component should render remove security group form 1`] = `
<Provider
  store={
    Object {
      "asyncReducers": Object {
        "FormButtons": [Function],
        "miqCustomTabReducer": [Function],
        "notificationReducer": [Function],
      },
      "dispatch": [Function],
      "getState": [Function],
      "injectReducers": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <AddRemoveSecurityGroupForm
    isAdd={false}
    recordId="1850"
    redirectURL="/vm_cloud/explorer"
  >
    <Loading
      active={true}
      className="export-spinner"
      description="Active loading indicator"
      small={true}
      withOverlay={false}
    >
      <div
        aria-atomic="true"
        aria-live="assertive"
        className="export-spinner bx--loading bx--loading--small"
      >
        <label
          className="bx--visually-hidden"
          id="loading-id-2"
        >
          Active loading indicator
        </label>
        <svg
          className="bx--loading__svg"
          viewBox="0 0 100 100"
        >
          <title>
            Active loading indicator
          </title>
          <circle
            className="bx--loading__background"
            cx="50%"
            cy="50%"
            r="44"
          />
          <circle
            className="bx--loading__stroke"
            cx="50%"
            cy="50%"
            r="44"
          />
        </svg>
      </div>
    </Loading>
  </AddRemoveSecurityGroupForm>
</Provider>
`;
